ProjFS.Mac/PrjFSKextTests/ProviderMessagingMock.cpp

```diff
@@ -9,10 +9,14 @@
 #include <libkern/OSAtomic.h>
 #include <sys/kauth.h>
 #include "ProviderMessagingMock.hpp"
+#include <functional>
+
+using std::function;
+using std::move;
 
 bool static s_defaultRequestResult = true;
 bool static s_secondRequestResult = true;
-bool static s_cleanupRootsAfterRequest = false;
+static function<void()> s_trySendRequestSideEffect;
```

Comment:
I haven't because `std::function` is a class and provides a default constructor which is appropriate to be used in this case. We could explicitly invoke it, but that syntax is rather wordy:

```
static function<void()> s_trySendRequestSideEffect = function<void()>();
```

I'm not convinced that's an improvement.